be/src/exec/pipeline/aggregate/spillable_aggregate_blocking_sink_operator.cpp

```diff
@@ -52,7 +59,23 @@ Status SpillableAggregateBlockingSinkOperator::set_finishing(RuntimeState* state
                 },
                 state, *io_executor, spill::MemTrackerGuard(tls_mem_tracker));
     };
-    RETURN_IF_ERROR(set_call_back_function(state, io_executor));
+
+    if (_aggregator->spill_channel()->is_working()) {
+        DCHECK(_spill_strategy == spill::SpillStrategy::SPILL_ALL);
+        std::function<StatusOr<ChunkPtr>()> task = [state, io_executor, flush_function,
+                                                    set_call_back_function]() -> StatusOr<ChunkPtr> {
+            RETURN_IF_ERROR(flush_function(state, io_executor));
```

Comment:
I'm not sure if flush_function return error. call back function works

Reply:
> I'm not sure if flush_function return error. call back function works

I have tested injecting errors in flush_function, which can introduce crashed. cancel is a common problem in spilling, I will fix it in the next PR